modules/stub/src/stub.js

```diff
@@ -1,241 +1,282 @@
 
-                } catch (ignore) {/* oh well... */}
+          setTimeout(addLocatorFrame, 5);
 
-              } else {
+        }
 
-                json = event.data;
+      }
 
-              }
+    };
 
-              const payload = json[TCF_POST_MSG_CALL];
+    addLocatorFrame();
 
-              if (payload) {
+    /**
+     * Listen for postMessage calls
+     */
 
-                const {command, version, callId, parameter} = payload;
+    const callMap = new Map();
+    const addEventListenerMap = new Map();
 
-                const wrapperCallback = (returnValue, success) => {
+    win.addEventListener(POST_MESSAGE_EVENT, (event) => {
 
-                  let returnMsg = {
-                    [TCF_POST_MSG_RETURN]: {
-                      returnValue,
-                      success,
-                      callId,
-                    },
-                  };
+      /**
+       * Make sure an event with a source and date and way to respond are
+       * available before doing anything with it.
+       */
 
-                  if (msgIsString) {
+      if (event && event.data) {
 
-                    /**
-                     * If it came in as a string it will be return in the same
-                     * manner
-                     */
+        if (typeof event.data === 'string') {
 
-                    returnMsg = JSON.stringify(returnMsg);
+          try {
 
-                  }
+            /**
+             * Try to parse the data from the event; there is no guarantee
+             * that this string is a stringified object – in fact it's
+             * probably more likely that it's not.
+             */
 
-                  event.source.postMessage(returnMsg, TARGET_ORIGIN);
+            event.data = JSON.parse(event.data);
 
-                };
+          } catch (ignore) {/* oh well... */}
 
-                win[TCF_API_NAME](
-                  command,
-                  version,
-                  wrapperCallback,
-                  parameter,
-                );
+        }
 
-              }
+        if (typeof event.data === 'object') {
 
-            }
+          if (isLeader && event.data[TCF_POST_MSG_CALL]) {
 
-          };
+            /**
+             * Since this instance is at the top, then the CMP exists here
+             * and we'll wrap all postMessage callbacks. Otherwise, this
+             * message should be ignored because this instance isn't tasked
+             * with handling it.
+             */
 
-          win.addEventListener(EVENT_NAME, postMessageEventHandler);
+            const {command, version, callId, parameter} = event.data[TCF_POST_MSG_CALL];
 
-        } else {
+            const wrapperCallback = (returnValue, success) => {
 
-          let callId = 0;
-          const callMap = new Map();
+              if (event.source && event.source.postMessage) {
 
-          win[TCF_API_NAME] = (command, version, callback, parameter) => {
+                event.source.postMessage({
```

Comment:
The v1.1 spec calls out that the response data should be the same type (string or object) as the message payload received.  The v2 spec does not call this out.  Has this changed?  Here, you are always returning an object.